carsus/io/output/tardis_.py

```diff
@@ -356,113 +379,160 @@ def lines(self):
-        whens.append((self.session.query(lvl_alias). \
-                     filter(and_(lvl_alias.atomic_number == Level.atomic_number,
-                                 lvl_alias.ion_charge == Level.ion_charge),
-                                 lvl_alias.data_source == self.ku_ds).exists(), self.ku_ds.data_source_id))
+        whens.append((
+            self.session.
+            query(lvl_alias).
+            filter(
+                and_(
+                    lvl_alias.atomic_number == Level.atomic_number,
+                    lvl_alias.ion_charge == Level.ion_charge),
+                lvl_alias.data_source == self.ku_ds
+                ).
+            exists(),
+            self.ku_ds.data_source_id
+            ))
 
         # 3. Else select from the nist source (the ground levels)
-        case_stmt = case(whens=whens, else_=self.nist_ds.data_source_id)
+        return case(whens=whens, else_=self.nist_ds.data_source_id)
+
+    def _build_levels_q(self):
 
-        levels_q = self.session.query(Level).\
-            filter(Level.atomic_number.in_(self.selected_atomic_numbers)).\
-            filter(Level.data_source_id == case_stmt)
+        levels_q = (
+                self.session.
+                query(Level.level_id).
+                filter(Level.atomic_number.in_(self.selected_atomic_numbers)).
+                filter(Level.data_source_id == self.data_source_case)
+                )
 
-        return levels_q
+        return levels_q.subquery()
 
-    def _build_lines_q(self, levels_ids):
-        levels_subq = self.session.query(Level.level_id.label("level_id")). \
-            filter(Level.level_id.in_(levels_ids)).subquery()
+    def _build_lines_q(self, levels_subq):
 
-        lines_q = self.session.query(Line). \
-            join(levels_subq, Line.lower_level_id == levels_subq.c.level_id)
+        lines_q = self.session.query(Line).join(
+                levels_subq,
+                Line.lower_level_id == levels_subq.c.level_id)
 
         return lines_q
 
-    @staticmethod
-    def _get_all_levels_data(levels_q):
-        levels = list()
-        for lvl in levels_q.options(joinedload(Level.energies)):
-            try:
-                energy = None
-                # Try to find the measured energy for this level
-                for nrg in lvl.energies:
-                    if nrg.method == "meas":
-                        energy = nrg.quantity
-                        break
-                # If the measured energy is not available, try to get the first one
-                if energy is None:
-                    energy = lvl.energies[0].quantity
-            except IndexError:
-                print "No energy is available for level {0}".format(lvl.level_id)
-                continue
-            levels.append(
-                (lvl.level_id, lvl.atomic_number, lvl.ion_charge, energy.value, lvl.g))
+    def _get_all_levels_data(self):
```

Comment:
quick documentation. just a quick statement of what this does. Do this as well for the other functions.